client/layout/index.jsx

```diff
@@ -87,6 +87,33 @@ function SidebarScrollSynchronizer( { enabled } ) {
 	return null;
 }
 
+function SidebarOverflowDelay( { layoutFocus } ) {
+	const setSidebarOverflowClass = ( overflow ) => {
+		const classList = document.querySelector( 'body' ).classList;
+		if ( overflow ) {
+			classList.add( 'is-sidebar-overflow' );
+		} else {
+			classList.remove( 'is-sidebar-overflow' );
+		}
+	};
+
+	React.useEffect( () => {
+		if ( layoutFocus !== 'sites' ) {
+			// Since `overflow` isn't an animated CSS property, we need to set it
+			// after the CSS transition of the layout elements finishes.
+			// @see https://github.com/Automattic/wp-calypso/blob/0c1176ddf16227307480fa4852e47cfe089bf181/client/layout/style.scss#L18
+			// @see https://github.com/Automattic/wp-calypso/issues/47019
+			setTimeout( () => {
+				setSidebarOverflowClass( true );
+			}, 500 );
```

Comment:
Can we align the timing `500` closer to the transition `150`?

Reply:
I tried it, but it seems that the transition doesn't start immediately after the current layout focus changes from `sites` to `sidebar`. When the timeout is set to `150`, the `overflow` property is set before the transition finishes.

I guess there is either a component async loaded or an async request that is delaying the start of the sidebar rendering. I couldn't identify where is happening that, so finally opted for extending the timeout. Going to add a comment clarifying that.